### PR TITLE
REST API: Remove XMLRPC check for site address login

### LINKED ISSUE #8770: REST API: Remove XMLRPC requirement for site credential login
- [ ] Add new config in `WordPressAuthenticator` to skip XMLRPC check for site address login
- [ ] Update the config in the app
- [ ] Handle login without XMLRPC API

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #8770
Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/736.

## Description

This PR removes the XMLRPC requirement to handle site credential login. Changes include:
- Added a new use case `SiteCredentialLoginUseCase`. This use case works similarly to `SiteCredentialViewModel`:
  - All cookies are cleared before handling login. This is important since existing cookies might cause the login to succeed even with incorrect site credentials.
  - Use the input credentials to handle cookie-nonce authentication.
  - Trigger a call to WP.org REST API to check if the authentication is successful. Here I'm using the plugin endpoint since it has already been implemented with cookie authentication.
  - The `SiteCredentialViewModel` will need to be updated to use this use case, but I'll handle that in a separate PR to avoid bloating this PR.
- Applied a new config `skipXMLRPCCheckForSiteAddressLogin` for the authenticator and implemented the delegate method `handleSiteCredentialLogin` to trigger `SiteCredentialLoginUseCase`.

Note: at the time of writing, we are forced to update WPKit version due to the requirement from the latest version of WPAuthenticator. Please follow the updates in an internal thread: p1675142099386409-slack-CC7L49W13.

## Testing instructions

- Update `enableSiteAddressLoginOnly` in `AuthenticationManager` to `true` to force the site address login only experience and build the app.
- On the prologue screen, select Enter your site address and proceed with the address of a self-hosted store.
- Enter incorrect site credentials - notice that an error is displayed for the incorrect credentials.
- Enter the correct credentials - the login should succeed and the app should navigate to the home screen.
- Notice in Xcode console that all Tracks event for the login flow should be sufficient for both success and failure cases, and no mention of XMLRPC should be present.

Please feel free to test again with `enableSiteAddressLoginOnly` disabled. Site address login should still use XMLRPC and succeed with sites that have working XMLRPC urls.

## Screenshots

N/A
---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/SiteCredentialLoginUseCase.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        // Old cookies can make the login succeeds even with incorrect credentials
+        // So we need to clear all cookies before login.
+        clearAllCookies()
+        loginAndAttemptFetchingJetpackPluginDetails(username: username, password: password)
+    }
+}
+
+private extension SiteCredentialLoginUseCase {
+    func clearAllCookies() {
+        let cookieJar = HTTPCookieStorage.shared
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/Authentication/SiteCredentialLoginUseCase.swift]: Nit: I think we might be able to unit test that cookies are cleared if we inject HTTPCookieStorage.

**Review by selanthiraiyan — APPROVED** [review]
This works as expected. 🚢

File: WooCommerce/Classes/Authentication/SiteCredentialLoginUseCase.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/Authentication/SiteCredentialLoginUseCase.swift]: Added test in cd774d85943a3d595b3cfb9cb599efff1d567900 and a2e59183a3065e6dfa604d8a6c09440e9bcfd5e8.

**Comment by itsmeichigo** [comment]
Pointed WPAuthenticator to `trunk` instead of releasing a beta version of the library following paaHJt-4lp-p2. Merging now.

**Pull request merged**